### PR TITLE
Improve performance of `db._explain()`

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

* Improve performance of `db._explain()` for very large query execution plans.
  Higher performance is achieved by not serializing some internal data
  structures when serializing execution plans. Serializing internal data is
  now opt-in and turned off if not needed. Apart from performance, there should
  be no end user-visible changes.

This is a performance-only improvement. No backports are planned.

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [x] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

### CODE REVIEW

**Review by dothebart — APPROVED** [review]

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,12 @@
 devel
 -----
 
+* Improve performance of `db._explain()` for very large query execution plans.
+  Higher performance is achieved by not serializing some internal data
+  structures when serializing execution plans. Serializing internal data is
+  now opt-in and turned off if not needed. Apart from performance, there should 
+  be no end user-visible changes.
```
Comment (cpjulia) [inline CHANGELOG:8]: wouldn't it be like end-user visible changes?

File: arangod/Aql/ExecutionPlan.cpp [hunk 2]
```diff
@@ -555,31 +555,40 @@ ExecutionPlan* ExecutionPlan::clone(Ast* ast) {
 /// @brief clone an existing execution plan
 ExecutionPlan* ExecutionPlan::clone() { return clone(_ast); }
 
+// build flags for plan serialization
+unsigned ExecutionPlan::buildSerializationFlags(
```
Comment (cpjulia) [inline arangod/Aql/ExecutionPlan.cpp:559]: is it useful for flags to become a member? e.g. _flags

**Review by cpjulia — APPROVED** [review]
LGTM

File: CHANGELOG (hunk 1 above)
Comment (jsteemann) [inline CHANGELOG:8]:
```suggestion
  be no end-user visible changes.
```

File: CHANGELOG [hunk 3]
```diff
@@ -1,6 +1,11 @@
 devel
 -----
 
+* Improve performance of `db._explain()` for very large query execution plans.
+  Higher performance is achieved by not serializing some internal data
+  structures when serializing execution plans. Serializing internal data is
+  now opt-in and turned off if not needed. Apart from performance, there should 
+  be no end-user visible changes.
 * Fixed BTS-712: Collation analyzer now always produces valid UTF-8 sequence.
```
Comment (jsteemann) [inline CHANGELOG:9]:
```suggestion
  be no end user visible changes.

* Fixed BTS-712: Collation analyzer now always produces valid UTF-8 sequence.
```

**Pull request merged**